src/utils/az.ts

```diff
@@ -36,4 +42,63 @@ export class Az {
 
     return { succeeded: true, result: subs };
   }
+
+  // this method considers getSubscriptions() is called first.
+  async getResourceGroups(subscription: Subscription): Promise<Errorable<ResourceGroup[]>> {
+    if (!(await this.azAccount.waitForLogin())) {
+      commands.executeCommand("azure-account.askForLogin");
+      return { succeeded: false, error: "failed to login" };
+    }
+    await this.azAccount.waitForFilters();
+    const resourceGrps: ResourceGroup[] = [];
+    for (const session of this.azAccount.sessions) {
+      const resources = new ResourceManagementClient(session.credentials2, subscription.subscriptionId!);
+      const resourceGrpPages = resources.resourceGroups.list().byPage();
+      for await (const page of resourceGrpPages) {
+        resourceGrps.push(...page);
+      }
+    }
+    return { succeeded: true, result: resourceGrps };
+  }
+
+  //this method considers getSubscriptions() and getResourceGroups() are called first
+  async getAksClusterNames(subscription: Subscription): Promise<Errorable<ManagedCluster[]>> {
+    const clusters: ManagedCluster[] = [];
+    for (const session of this.azAccount.sessions) {
+      const client = new ContainerServiceClient(session.credentials2, subscription.subscriptionId!);
+      for await (const item of client.managedClusters.list().byPage()) {
+        clusters.push(...item);
+      }
+    }
+    return { succeeded: true, result: clusters };
+  }
+
+  async getAcrRegistries(endpoint: string): Promise<Errorable<string[]>> {
```

Comment:
we also need a function to get the endpoint of the registry list acr endpoints of something